12/11/14
===

Distances
---

Work on the [code](https://github.com/pilarcormo/small_genomes_SNPs/blob/master/Make_perm_shorter.rb) that takes the last permutation of the run and gradually removes the first and last contig to compare the remaining order with the correctly ordered genome to see if the distance measurements improves when comparing the contigs that are in the middle of the permutation. 

Same problem than yesterday. 


Finding the causative SNP mutation with my small genomes v 3.0
---
Run find_causal_mutation.rb with small genomes I created last month. 
 
â€¢__2 Kb genome using count ratio method__: 

Position in the correct ordered genome: 997


<table>

  <tr><th>Fitness method</th><th>Number of generations</th><th>Population size</th><th>Select number</th><th>Chunk mutants</th><th>Swap mutants</th><th>Save</th><th>Random</th><th>Divisions (1000s)</th><th>Location SNP</th><th>Name of the file</th></tr>
  
  
  <tr> <td>count ratio</td> <td>150</td> <td>10</td> <td>5</td> <td>7</td> <td>7</td> <td>5</td> <td>2</td> <td>10</td><td>1580</td><td>count_ratio</td> </tr>
  <tr> <td>count ratio</td>  <td>150</td> <td>100</td> <td>50</td> <td>35</td> <td>35</td> <td>25</td> <td>5</td> <td>5</td> <td>1015</td><td>count_ratio4</td></tr>
10
    <tr> <td>count ratio</td>  <td>150</td> <td>100</td> <td>50</td> <td>35</td> <td>35</td> <td>25</td> <td>5</td> <td>10</td> <td>1421</td><td>count_ratio2</td></tr>
    <tr> <td>count ratio</td>  <td>150</td> <td>100</td> <td>50</td> <td>35</td> <td>35</td> <td>25</td> <td>5</td> <td>100</td> <td>687</td><td>count_ratio3</td></tr>
       
  
</table>
 

Create new model genomes
---

The chromosome 4 from Arabidopsis is 18 Mb long, so when using it as a model genome, the algorithm runs very slowly. We propose to use the algorithm with increasingly larger genomes to improve the algorithm performance getting closer to a model that can be extrapolated to reality. 

Based on the following script (small_test_genome):

```
hm_r = 'hm <- rnorm(50, 1000, 100)' # Causative SNP at/near 1000
ht_r = 'ht <- runif(50, 1, 2000)'   # Genome length of 2000
hm, ht = ModelGenome::get_snps(hm_r, ht_r)
snp_pos = [hm, ht].flatten

arabidopsis_c4 = ModelGenome::fasta_to_char_array("TAIR10_chr4.fasta")
small_genome = arabidopsis_c4[-8000..-1] # Genome length of 8Kb

contig_size = 25 # 25-50 bp
frags = ModelGenome::get_frags(small_genome, contig_size)

```

<table>
  <tr><th>Size of the genome (kb))</th><th>From</th><th>Contig size</th><th>rnorm</th><th>runif</th><th>Name of the file</th></tr>
   <tr><th>8</th><th>arabidopsis_c4[1..8000]<th>50</th></th><th>runif(100, 1, 8000)</th><th>runif(100, 1, 8000)</th><th>8kbgenome</th></tr>
   <tr><th>8</th><th>arabidopsis_c4[-8000..-1]<th>50</th></th><th>rnorm(50, 4000, 200)</th><th>unif(50, 1, 8000)</th><th>8kbgenome2</th></tr>
    <tr><th>20</th><th>arabidopsis_c4[-20000..-1]</th><th>200</th><th>rnorm(200, 10000, 200)</th><th>runif(200, 1, 20000)</th><th>20kbgenome</th></tr>
     
     <tr><th>100</th><th>arabidopsis_c4[-100000..-1]</th><th></th><th>rnorm(100, 50000, 5000)</th><th>runif(100, 1, 100000)</th><th>100kbgenome</th></tr>
     
      <tr><th>1000</th><th</th><th></th><th></th><th></th><th></th></tr>
 </table>
 
 
####8kbgenome output
There are 94 homozygous SNPs

There are 99 heterozygous SNPs

Is there a SNP at the centre of the distribution? -- false

You have created 107 fragments of sizes 50-100

####8kbgenome2 output
There are 48 homozygous SNPs

There are 50 heterozygous SNPs

Is there a SNP at the centre of the distribution? -- false

You have created 110 fragments of sizes 50-100

####20kbgenome output
There are 171 homozygous SNPs

There are 199 heterozygous SNPs

Is there a SNP at the centre of the distribution? -- false

You have created 65 fragments of sizes 200-400
 