14/11/14
===

Distances
---

Work on the [code](https://github.com/pilarcormo/small_genomes_SNPs/blob/master/Make_them_shorter.rb) that takes the last permutation of the run and gradually removes the first and last contig to compare the remaining order with the correctly ordered genome to see if the distance measurements improves when comparing the contigs that are in the middle of the permutation. 

Now, if the item we are looking for is present in the permutation, we proceed normally, but if it is not, we add a different value. **I've chosen the length of the orignal array as the value to add just to try how it works, but not sure if this is the correct approach I should take**. I've also tried with 0, 1. 

By now, I'm just focusing on the deviation distance method. 

####New Idea
We don't need to take out the first and last element in the original array as well. We need to remove the values we are taking out of the permutation. 

#####Example:

*Original (1, 2, 3, 4, 5, 6)*

*Permutation (2, 3, 4, 5, 6, 1)*

Remove first and last items **from permutation** and the same items from the original array:

*Permutation (<s>2</s>, 3, 4, 5, 6, <s>1</s>)*

*Original (<s>1</s>, <s>2</s>, 3, 4, 5, 6)*

IT WORKS!

```
while i < length/2 do
			left = i
			right = (length-1) - i
			p = perm[left..right]
			perm.each do |a|
				unless p.include?(a)
					original.delete(a)
				end
			end
			
```
2 kb genome

Maximum density
image

Count ratio (from countratio2)
image

20 kb genome
---
Using the count ratio method:

Location of causal mutation in correctly ordered genome: 10028

Candidate SNP position in permutation: 11239