03-04/08/15
===

- Work on 3.4 
- run SDM on 2-5 kb contigs genomes. mob1 and BCF2 worked perfectly and sup#1 was ok. However, in mob1 and OCF2, the position of the mutation was missing.

Finally, I have developed an automatic way to filter the contigs based on the % of maximum ratio as depending on the datasets, sometimes the majority of the contigs are discarded at 10% while in other examples, a 30% is required to get a good level of filtering. Let's say 1% is used as default, if the contigs discarded are less than 5% of the original contig number, the filtering step is repeated with a threshold of 2 times the default (2% in this example)  

```
while ids.length > 20*contigs_discarded do 
	threshold = threshold*2
	puts "filtering too low, increase to #{threshold} %"
	dic_ratios, ratios, ids_s, dic_ratios_inv  = Ratio_filtering.important_ratios(snps_hm, snps_ht, ids, threshold, adjust)  
	contigs_discarded = ids.length - ids_s.length
	puts "#{contigs_discarded} contigs out of #{ids.length} discarded"
end 
```

The contig size has a strong effect on the number of candidate contigs required when using real densities. Before, when we used 10-100 kb contigs, the total number of contigs was lower but the SNP density was the same, so fewer contigs were needed as candidates to identify the mutation (more SNPs per each contig). Now, for small contigs (2-5 kb), more candidate contigs are required (as the total number is also larger). Therefore, I needed to define a new constant (average_contig) and depending on its value, 


```
    if average_contig.to_i < 10000
      if cross == "back" #In case of a back-cross, 20 contigs in the middle part of the permutation taken
        if right.length > 10
          mut << right.flatten[-10, 10]
          mut << left.flatten[-10, 10].reverse
          mut.flatten!
        else 
          mut << right.flatten[-right.length, right.length]
          mut << left.flatten[-left.length, left.length].reverse
          mut.flatten!
        end 
     elsif cross == "out" #In case of a out-cross, 50 contigs in the middle part of the permutation taken
        if right.length > 25
          mut << right.flatten[-25, 25]
          mut << left.flatten[-25, 25].reverse
          mut.flatten!
        else 
          mut << right.flatten[-right.length, right.length]
          mut << left.flatten[-left.length, left.length].reverse
          mut.flatten!
        end 
      end 
    else ###contig size over 10 kb
      mut << right.flatten[-5, 5]
      mut << left.flatten[-5, 5].reverse
      mut.flatten!
    end 
 ```
 
05/08/15
===

- Try new automatic ratio with all the contig sizes.
 
Next
---

 Maybe table with all the options
- Change table with new results
- Finish contig N50 plot
- Travis CI
