13/10/14
========================================================


Testing the Genetic Algorithm To Order Contigs (GATOC) using a small model genome
-------

1. I create 2 small model genomes of **2 kb and 8 kb** based on [Arabidopsis chromosome 4](https://github.com/pilarcormo/fragmented_genome_with_snps/blob/master/TAIR10_chr4.fasta) by using the [small_test_genome.rb](https://github.com/pilarcormo/fragmented_genome_with_snps/blob/master/small_test_genome.rb) script. This small genome is fragmented into contigs of **25 bp**. The files [dataset_small2kb](https://github.com/pilarcormo/fragmented_genome_with_snps/tree/master/arabidopsis_datasets/Pilar/dataset_small2kb) and [dataset_small8kb](https://github.com/pilarcormo/fragmented_genome_with_snps/tree/master/arabidopsis_datasets/Pilar/dataset_small8kb) are created as output. 


2. For each of the genomes, I run the [algorithm_test.rb](https://github.com/pilarcormo/fragmented_genome_with_snps/blob/master/algorithm_test.rb) with the following parameters:

<table>

  <tr><th>Fitness method</th><th>Number of generations</th><th>Population size</th><th>Select number</th><th>Chunk mutants</th><th>Swap mutants</th><th>Save</th><th>Random</th><th>Divisions (1000s)</th></tr>
  
  
  <tr> <td>snp_distance</td> <td>10</td> <td>100</td> <td>50</td> <td>35</td> <td>35</td> <td>25</td> <td>5</td> <td>1</td> </tr>
<<<<<<< HEAD
    <tr> <td>max_density</td>  <td>10</td> <td>100</td> <td>50</td> <td>35</td> <td>35</td> <td>25</td> <td>5</td> <td>1</td> </tr>
    <tr> <td>max_ratio</td>  <td>10</td> <td>100</td> <td>50</td> <td>35</td> <td>35</td> <td>25</td> <td>5</td> <td>1</td> </tr>
=======
  <tr> <td>max_density</td>  <td>10</td> <td>100</td> <td>50</td> <td>35</td> <td>35</td> <td>25</td> <td>5</td> <td>1</td> </tr>
  <tr> <td>max_ratio</td>  <td>10</td> <td>100</td> <td>50</td> <td>35</td> <td>35</td> <td>25</td> <td>5</td> <td>1</td> </tr>
<<<<<<< HEAD
>>>>>>> FETCH_HEAD
=======
>>>>>>> FETCH_HEAD
  
</table>
 
