22/06/15
===

From NCBI I selected all plant genome assemblies at contig level from 2013 until now. [contigs.md](https://github.com/pilarcormo/SNP_distribution_method/blob/master/Contigs/contigs.md)

Characteristics: 

- full genome representation 
- genome coverage over 1x
- sequencing technology
- contig N50 length. 

I got 30 assemblies in plants. From those, 15 used Illumina HiSeq, so this is the preferred technology. To analyse the contig size distribution, I divide the genome length by the number of contigs to get an average contig size. I also used the N50 contig length. 

Write and run [contigs.R]()

Use Inkscape to redo figures for preprint. 


23/06/15
===

- Statistics: number of assemblies for each technology, N50 contig size distribution (median, mean, quartiles). Influence of genome coverage. Plots are in [Contigs](https://github.com/pilarcormo/SNP_distribution_method/tree/master/Contigs).



24/06/15
===
-  Specify threshold value as argument instead of using always max.ratio/4. Now I use max.ratio divided by 100, 50, 10 and 5.

-  Contig size analysis. Try plotting probabilities of contig sizes for each genome length. Not sure how. 


###Next

- Table with all the studies 
- Workflow for SDM, workflows in methods instead of results. 
- Provide scripts or commands used in methods
- Incorporate contigs analysis? 
