20/07/15
===
- write piece of code in SNP_distribution_method.rb in Small_genomes/ to create csv files with deviations 
-  Write density_snp.rb to create csv file with all the SNP densities before filtering, after parental filtering and after removing centromeres. Maybe make a plot for additional results for preprint. However:	
	- Some filtering was missing in Reads/ - done it again
	
	- sup1 had a mistake in the snp density files (they were all the same for all the chromosomes), I am re-doing all the filtering: 

```
for i in {1..5} 
do
	ruby manage_vcf.rb Reads/Aw_sup1-2 sup1.vcf sup1_chromosome$i $i cutting_vcf
done 
for i in {1..5} 
do
	ruby manage_vcf.rb Reads/Aw_sup1-2 colT.vcf Parental/colT_chromosome$i $i cutting_vcf
done 
for i in {1..5} 
do
	ruby manage_vcf.rb Reads/Aw_sup1-2 WsT.vcf Parental/Ws_chromosome$i $i cutting_vcf
done 
for i in {1..5} 
do
	ruby manage_vcf.rb Reads/Aw_sup1-2 sup1_chromosome$i/chromosome$i.vcf filter1_chromosome$i $i filter_vcf Parental/Ws_chromosome$i/chromosome$i.vcf
done 
for i in {1..5} 
do
	ruby manage_vcf.rb Reads/Aw_sup1-2 filter1_chromosome$i/chromosome$i.vcf filter2_chromosome$i $i filter_vcf Parental/ColT_chromosome$i/chromosome$i.vcf
done 
```
I added this to methods in Code.md

21/07/15
===
- Repeat sup1 model genomes 
bsub -q TSL-Test128 -o nocen.txt "ruby model_genome_new.rb new_sup1_nocen_chr4_10kb 10000 new_snps/filter2_chromosome4 4"
bsub -q TSL-Test128 -o nocen.txt "ruby model_genome_new.rb new_sup1_nocen_chr4_100kb 100000 new_snps/filter2_chromosome4 4"
- Code.md -> sup_file1.md (supplementary file with command lines for read mapping and snp calling)
- sup_file2.md (supplementary file with command lines for filtering, etc)
- density.csv - sum all the densities per screen and plot





