20/07/15
===
- write piece of code in SNP_distribution_method.rb in Small_genomes/ to create csv files with deviations 
-  Write density_snp.rb to create csv file with all the SNP densities before filtering, after parental filtering and after removing centromeres. Maybe make a plot for additional results for preprint. However:	
	- Some filtering was missing in Reads/ - done it again
	
	- sup1 had a mistake in the snp density files (they were all the same for all the chromosomes), I am re-doing all the filtering: 

```
for i in {1..5} 
do
	ruby manage_vcf.rb Reads/Aw_sup1-2 sup1.vcf sup1_chromosome$i $i cutting_vcf
done 
for i in {1..5} 
do
	ruby manage_vcf.rb Reads/Aw_sup1-2 colT.vcf Parental/colT_chromosome$i $i cutting_vcf
done 
for i in {1..5} 
do
	ruby manage_vcf.rb Reads/Aw_sup1-2 WsT.vcf Parental/Ws_chromosome$i $i cutting_vcf
done 
for i in {1..5} 
do
	ruby manage_vcf.rb Reads/Aw_sup1-2 sup1_chromosome$i/chromosome$i.vcf filter1_chromosome$i $i filter_vcf Parental/Ws_chromosome$i/chromosome$i.vcf
done 
for i in {1..5} 
do
	ruby manage_vcf.rb Reads/Aw_sup1-2 filter1_chromosome$i/chromosome$i.vcf filter2_chromosome$i $i filter_vcf Parental/ColT_chromosome$i/chromosome$i.vcf
done 
```
I added this to methods in Code.md

21/07/15
===
- Repeat sup1 model genomes 

```
bsub -q TSL-Test128 -o nocen.txt "ruby model_genome_new.rb new_sup1_nocen_chr4_10kb 10000 new_snps/filter2_chromosome4 4"
bsub -q TSL-Test128 -o nocen.txt "ruby model_genome_new.rb new_sup1_nocen_chr4_100kb 100000 new_snps/filter2_chromosome4 4"
```

- Code.md -> sup_file1.md (supplementary file with command lines for read mapping and snp calling)
- sup_file2.md (supplementary file with command lines for filtering, etc)
- density.csv - sum all the densities per screen and plot

22/07/15
===

- Run SDM on sup1_nocen_chr4_10kb and sup1_nocen_chr4_100kb. 
- Write command lines for SDM on model genomes
- Try again to fit a model to contig distribution. Decide on figures for preprint
- Finish methods?
- Update figures


23-24/07/15
===

- Proofread methods
- Write mutation candidate positions for 10kb contigs. Rerun ratios.sh (I add a file "hyp_positions.txt" to include the hypothetical positiosn in a file. In this case it can be known since we know the correct contig order. )

```
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/ocf2_nocen_chr2_10kb ocf2_SDM_0 0 1 out ###filtering step off
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/ocf2_nocen_chr2_10kb ocf2_SDM_1 1 1 out
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/ocf2_nocen_chr2_10kb ocf2_SDM_2 2 1 out
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/ocf2_nocen_chr2_10kb ocf2_SDM_10 10 1 out
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/ocf2_nocen_chr2_10kb ocf2_SDM_20 20 1 out
```

```
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/bcf2_nocen_chr3_10kb bcf2_SDM_0 0 1 back ###filtering step off
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/bcf2_nocen_chr3_10kb bcf2_SDM_1 1 1 back
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/bcf2_nocen_chr3_10kb bcf2_SDM_2 2 1 back
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/bcf2_nocen_chr3_10kb bcf2_SDM_10 10 1 back
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/bcf2_nocen_chr3_10kb bcf2_SDM_20 20 1 back
```
```
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/B_nocen_chr5_10kb b_SDM_0 0 1 back ###filtering step off
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/B_nocen_chr5_10kb b_SDM_1 1 1 back
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/B_nocen_chr5_10kb b_SDM_2 2 1 back
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/B_nocen_chr5_10kb b_SDM_10 10 1 back
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/B_nocen_chr5_10kb b_SDM_20 20 1 back
```
```
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/C_nocen_chr5_10kb c_SDM_0 0 1 back ###filtering step off
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/C_nocen_chr5_10kb c_SDM_1 1 1 back
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/C_nocen_chr5_10kb c_SDM_2 2 1 back
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/C_nocen_chr5_10kb c_SDM_10 10 1 back
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/C_nocen_chr5_10kb c_SDM_20 20 1 back
```

```
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/sup1_nocen_chr4_10kb sup1_SDM_0 0 1 out ###filtering step off
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/sup1_nocen_chr4_10kb sup1_SDM_1 1 1 out
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/sup1_nocen_chr4_10kb sup1_SDM_2 2 1 out
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/sup1_nocen_chr4_10kb sup1_SDM_10 10 1 out
ruby SNP_distribution_method_variation.rb arabidopsis_datasets/No_centromere/sup1_nocen_chr4_10kb sup1_SDM_20 20 1 out
```

- Write SDM.sh

- Change SDM.rb. Notised that when using 10kb contigs, 3 contigs at each side of the distribution peak are enough to contain the mutation in out-cross and back-cross experiments. Try with 2 at each side but was not enough for some of them (sup1 and mob2). New outputs are in folders starting with ---.

```
    if cross == "back" #In case of a backcross, 6 contigs in the middle part of the permutation taken
      mut << right.flatten[-3, 3]
      mut << left.flatten[-3, 3].reverse
      mut.flatten!
```

So probably I'm going to eliminate the back/out specification as argument.

###NEXT

- Results - rewrite, new results to discuss (contigs, SNP density analysis, ratio as threshold is not that crucial)
- Rewrite README files if I finally decide to remove back/out 