01/12/14
=====

- Use cluster to create new datasets and new genomes. 



<table>

  <tr><th>Name</th><th>ht SNPs</th><th>hm SNPs</th><th>Number of contigs</th></tr>
  
  <tr> <td>"size"Mb</td> <td>1/1000</td> <td>constant</td> <td>700</td>  </tr>
  <tr> <td>A_"_size"Mb</td> <td>1/500 </td> <td>constant </td> <td>700 </td>  </tr>
  <tr> <td>B_"size"Mb</td> <td>1/500 </td> <td>constant </td> <td> 1300</td>  </tr>
  <tr> <td>C_"size"Mb</td> <td>1/1000 </td> <td> constant</td> <td> 1300</td>  </tr>
  <tr> <td>D_"size"Mb</td> <td>1/2000 </td> <td> constant</td> <td> 1300</td>  </tr>
  <tr> <td>E_"size"Mb</td> <td>1/2000 </td> <td>constant </td> <td> 700</td> </tr>
</table>

02/12/14
======

- Build and analyse final plots. 

```
z <- ggplot (z, aes (Genome.length, Divisions, fill = Fitness)) + geom_tile() + labs(title = "1 ht SNP/2000 bp
+ 700 contigs")

g_legend<-function(a.gplot){
	tmp <- ggplot_gtable(ggplot_build(a.gplot))
	leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
	legend <- tmp$grobs[[leg]]
	return(legend)}
	
mylegend <- g_legend(f)

multiplot <- grid.arrange(arrangeGrob(f + theme(legend.position="none"),g + theme(legend.position="none"), h + theme(legend.position="none"), ncol = 3), mylegend, ncol = 2)
 
 ```
 

- New genomes, hm SNP number = ht SNP number
<table>

  <tr><th>Name</th><th>ht SNPs</th><th>hm SNPs</th><th>Number of contigs</th></tr>

<tr> <td>F_"size"Mb</td> <td>1/2000 </td> <td>1/2000</td> <td> 700</td> </tr>
</table>



 
03/12/14
======

- New genomes, hm SNP number = ht SNP number
- Analyse the effect of increasing the number of contigs in the fitness
<table>

  <tr><th>Name</th><th>ht SNPs</th><th>hm SNPs</th><th>Number of contigs</th></tr>

<tr> <td>F_"size"Mb</td> <td>1/2000 </td> <td>1/2000</td> <td> 700</td> </tr>
<tr> <td>G_"size"Mb</td> <td>1/2000 </td> <td>1/2000</td> <td> 1300</td> </tr>
<tr> <td>H_"size"Mb</td> <td>1/2000 </td> <td>1/2000</td> <td> 3000</td> </tr>
</table>

04/12/14
======
- Analyse the effect of increasing the SNP density. 

<table>

  <tr><th>Name</th><th>ht SNPs</th><th>hm SNPs</th><th>Number of contigs</th></tr>
<tr> <td>F_"size"Mb</td> <td>1/2000 </td> <td>1/2000</td> <td> 700</td> </tr>
<tr> <td>I_"size"Mb</td> <td>1/1000 </td> <td>1/1000</td> <td> 700</td> </tr>
<tr> <td>J_"size"Mb</td> <td>1/500 </td> <td>1/500</td> <td> 700</td> </tr>

</table>

05/12/14
======
- Change  plots colour: 

``` 
> colors = brewer.pal(9, "BrBG")
> pal <- colorRampPalette(colors)
> f <- f + fill_gradn(pal)
```
- Create new genomes with the new number of swaps that seem to give better results

<table>

  <tr><th>Name</th><th>ht SNPs</th><th>hm SNPs</th><th>Number of contigs</th></tr>
<tr> <td>AA_"size"Mb</td> <td>1/500 </td> <td>1/1000</td> <td> 700</td> </tr>
<tr> <td>BB_"size"Mb</td> <td>1/1000 </td> <td>1/1000</td> <td> 700</td> </tr>
<tr> <td>CC_"size"Mb</td> <td>1/2000 </td> <td>1/1000</td> <td> 700</td> </tr>

</table>

- Repeat genomes G and H, because the number of swaps mutants shouldn't be the same than in the 700 contigs, if we have more contigs, I should increase in the same proportion the number of swappings. 


<table>

  <tr><th>Name</th><th>ht SNPs</th><th>hm SNPs</th><th>Number of contigs</th><th>Swaps</th></tr>
<tr> <td>GG_"size"Mb</td> <td>1/2000 </td> <td>1/2000</td> <td> 1300</td> <td>260</td></tr>
<tr> <td>HH_"size"Mb</td> <td>1/2000 </td> <td>1/2000</td> <td> 3000</td><td> 600</td> </tr>


</table>

Didn't have time to finish: HH genomes and CCtables. 
