08/12/14
=====

Finish CC plots and polish [results](https://github.com/pilarcormo/Correlations/blob/master/Results/Results.md). 

Now: run the algorithm with the parameters that seem to work fine as seen in the correlation experiment. First, assign SNPs distributions. 

09/12/14
=====
Create model genome using the [chromosome 1 from A. thaliana](http://www.ebi.ac.uk/ena/data/view/CP002684&display=fasta) of lenght 30427671 bp


There are 9988 homozygous SNPs
There are 9995 heterozygous SNPs

1690 contigs are created (ideally, this number should be higher, because we are generating contigs that are too large.)


Prepare sh to run algorithm in cluster.


10/12/14
=====

Run yesterday in the cluster failed (run"i") because the R package 'pracma' needed in the GATOC script was not installed. I was using the NBI cluster, so I change to the TSL cluster and try running them again (run"i"_letter) as it seems the 'pracma' library is installed there. 

```
bsub -q TSL-Prod128 -o 1.txt "ruby algorithm_test.rb 1_A run1_A count_ratio 1000 20 10 7 7 5 1 10 false"

bsub -q TSL-Prod128 -o 2.txt "ruby algorithm_test.rb 1_A run2_A count_ratio 1000 20 10 7 7 5 1 100 false "

bsub -q TSL-Prod128 -o 3.txt "ruby algorithm_test.rb 1_A run3_A count_ratio 1000 20 10 7 7 5 1 200 false"

```

11/12/14
=====

12/12/14
=====