19/01/15
====

- Continue testing new method. For now, I am going to use only the method to order the contigs based on the homozygous SNP distribution. 

- Create **new genomes** by re-writing the model genome script. 

- In order to create replicates for the experiment to be reliable, I am going to create 5 identical genomes for each condition. I am going to test the effect of different sizes from **1 to 15 Mb and different contig sizes**. The folders without letter contain genomes divided in 700 contigs and the folders with an "A" contain genomes with 1300 contigs. For the SNP distribution, I have chosen a high density, **1 SNP per 500 bp** (same number for hm and ht). If this works, I may reduce the SNP density. 

- Once I have the genomes in the share folder in the HPC, I run the SDM to create the ordered fasta file. After that, I am adapting the **find_causal_mutation.rb** script in order to create a common **CSV file** that contains all the different candidate SNP positions for all the genomes and the percentage of shift (how far is the candidate SNP from the "real" causal mutation).

20/01/15
====

- Run the script to find the **causal mutation** in cluster for all the genomes I created yesterday. 

For each genome size, there are a few replicates that show a great shift between the real location and the candidate SNP location. Its always around 48% what is weird because it does not seem random but instead we may say that the candidate SNP position has been multiplied by 2.  Trying to run SDM individually for each genome but don't know why it is not working

- TSL seminar
- Speak like a leader boot camp. 


#####Next
- **Plot results** to see if there is any correlation between the genome size, the number of contigs and the location of the mutation. For that, calculate the arithmetic mean for the 5 replicates created for each genome size. 

- Try to build the **distribution plots** for the SNP densities obtained after running the SDM.    

21/01/15
====

Identify problem when using the method with some genomes, in which the location of the causal mutation was moved to the end of the genome. The problem was in the method I wrote to look for the minimum values of SNP density. I think I've fixed it although the code is more complicated. 

- Run the method one more time. Output in folder Perm3. They seem alright now. 
- Plot results!
![Image](https://github.com/pilarcormo/small_genomes_SNPs/blob/master/Results/bar_chart.png?raw=true)

22/01/15
====

- Use SDM with the chromosome 1 (using different contig sizes) in cluster to see if the results obtained are reproducible with larger genomes (still running).
- Write test units for the code I have. Start by making new clases from SDM to make it look cleaner. 

23/01/15
====

I create several genomes using the whole chromosome 1 from A.thaliana. For each contig size, I created 5 replicates and calculated the average of the shift from the correct SNP location:


<table>
  <tr><th>Number of contigs <th>%</th>
  
  <tr><th>1015.2 </th> <th> 0.525776685</th>
  <tr><th> 2028.8 </th> <th> 0.426776667</th>
  <tr><th> 4055</th> <th>0.505681983</th>
</table>

The contig size has not a significant effect on the performance of SDM and the method is useful with larger genomes as shown here. 

- Write unit tests for SDM and Stuff. Works perfectly. 