26/01/15
===

We've been using normal distributions for the homozygous SNP densities. Therefore, in all our model genomes the causative mutations was always in the middle of the genome. But, **what happens if the mutation is at the beginning or the end of the model genome? Will the SDM work in that case?** This method is effective when our causative mutation is in the middle of distribution, but not when it is moved towards the ends. 

Tried that with the whole chromosome 1 in the cluster while doing some trials with the 2kb genome in the shell. 

i = 1..5
<table>
 <tr><th>Name <th>Contig length</th> <th>Mean of rnorm distribution</th>
 <tr><th>chr1_C_i <th>10000 </th> <th>20000000</th>
 <tr><th>chr1_D_i <th>10000 </th> <th>30000000</th>
 <tr><th>chr1_E_i <th>10000 </th> <th>10000000</th>
   
</table>

Reduce number of hm SNPs while keeping the same the number of ht SNPs. 

Create new genomes from 1 to 15 Mb with this distributions:

```
hm_r = "hm <- rnorm(#{snp}, #{size/3}, #{snp*2})" 

ht_r = "ht <- runif(#{snp}, 1, #{size})"   
```

where the variable 'snp' is defined as 2 times the size of the whole genome divided by 1000. 