3/02/15
====

Working on BCF2 in cluster. 

```
bsub -q TSL-Test128 -o sam.txt "bwa mem TAIR10_chr_all.fas BC.fg.reads1.fq BC.fg.reads2.fq > BCF2.sam"
bsub -q TSL-Test128 -o bcf2.txt "view -bS -h BCF2.sam > BCF2.bam"
bsub -q TSL-Test128 -o bcf2.txt "samtools mpileup -B -f TAIR10_chr_all.fas BCF2.bam > BCF2.pileup"
```

When counting lines in BAM file (```view BCF2.bam | wc -l```), got this error:

```
[W::sam_read1] parse error at line 1
[main_samview] truncated file.
0

```

SAM file is wrong. 

Repeat BWA alignment using default parameters. 

- Rstudio to plot kernel densities for hm SNPs per chromosome from[Scheneeberger](http://1001genomes.org/software/shoremap.html) reads.

```
density <- ggplot(hm) + geom_density(aes(V1), colour ="blue") + xlim(0,30000000) + xlab("Genome length") + ylab("SNP density") + theme_bw()
```

For ```xlim```, I use the chromosomes sizes from [here](https://www.arabidopsis.org/portals/genAnnotation/gene_structural_annotation/agicomplete.jsp)

New model genome from chr1
---
- Fragment chromosome 1 from Arabidopsis as usually. Create 10,000-20,000 contigs. 
- Instead of using a normal distribution for hm SNPs and a uniform distribution to define the ht SNPs, I use the files hm.txt and ht.txt obtained from the VCF file from real data. 

Write [results with real SNP densities](https://github.com/pilarcormo/SNP_distribution_method/blob/master/Results/Results_NGSdata.md)


4/02/15
====
- R online course. 1st class

pileup file from BCF2 is fine, download VarScan and obtain VCF as output:

```
bsub -q TSL-Test128 -o BCF2.txt "java -jar VarScan.v2.3.7.jar mpileup2snp BCF2.pileup --output-vcf 1"  
```

**14222 variant positions reported (14222 SNP, 0 indel)**

New model genome from chr1
---
After running in the cluster, the VCF file obtained is wrong (AF=WRONG in all the SNPs reported). The problem was that I was using the method file_to_array instead of file_to_ints_array from Write_it so the positions were not determined properly. Fixed it and running the model_genome script again in the cluster


5/02/15
====

Problems with VCF output from VarScan. It appears in the output.txt file from the job submission but not as an intedependent file. After taking it out from there, it was not recognised as a vcf file and when running the manage_vcf script, always got this error: 
```
VCF lines must have at least 8 fields (RuntimeError)
```
Tried to fix it by uploading the file in Galaxy and downloading it as a vcf, count number of columns, eliminate parts of the file... The error was a warning message in the middle of the file. Now it's fixed and I got all the files with the hm and ht SNP densities per chromosome. 


```
bsub -q TSL-Test128 -o BCF2_real.txt "ruby model_genome_new.rb BCF2_1 10000 1"
bsub -q TSL-Test128 -o BCF2_real.txt "ruby model_genome_new.rb BCF2_2 10000 2"
bsub -q TSL-Test128 -o BCF2_real.txt "ruby model_genome_new.rb BCF2_3 10000 3"
bsub -q TSL-Test128 -o BCF2_real.txt "ruby model_genome_new.rb BCF2_4 10000 4"
bsub -q TSL-Test128 -o BCF2_real.txt "ruby model_genome_new.rb BCF2_5 10000 5"
```