16/03/15
===
- Use trimmomatic to trim the BCF2 reads. 
- Rakefile
- Use [Galvao, 2012 and Schneeberger, 2009](http://1001genomes.org/software/shoremap.html) reads to obtain new SNP densities. 
- Take the hm/ht ratio into account in SDM. Normally, in the centromere, there is high hm and ht SNP densities, so we'll obtain a low ratio in that area. 


###Change SDM 
- I have defined the ratios before running SDM and discarded the contigs with low hm/ht ratio before trying to order them based on the hm SNP density. It seems to work better. 
