16/03/15
===
- Use trimmomatic to trim the BCF2 reads. 
- Rakefile
- Use [Galvao, 2012 and Schneeberger, 2009](http://1001genomes.org/software/shoremap.html) reads to obtain new SNP densities. 
- [Galvao paper](http://onlinelibrary.wiley.com/doi/10.1111/j.1365-313X.2012.04993.x/full#ss9). In this experiment, they induced by EMS a recessive mutation that causes a late flowering. They backcross the mutant with Ler and sequence. 
- Take the hm/ht ratio into account in SDM. Normally, in the centromere, there is high hm and ht SNP densities, so we'll obtain a low ratio in that area. 


######Change SDM 
- I have defined the ratios before running SDM and discarded the contigs with low hm/ht ratio before trying to order them based on the hm SNP density. It seems to work better. 

17/03/15
===

- Running Galvao alignments 
- Continue trimming BCF2 -> after removing first 10 bases (kmer content), FastQC still shows a wrong kmer content (with ~24 wrong bases at the beginning). It still appears as wrong after removing the 24 bp (+10 before) so I am leaving it as it was, because I am not making the reads shorter. 
- In SDM, create real hm and ht plots. Instead of using the original files for the hm and ht positions, I removed the positions where the ratio is low and try to build the plots with these new densities to have the shorten expected distributions. 
- Find out problem with rakefile in the cluster. As the task to do is to bsub the job in the cluster, once the submission is done, it moves to the next task even though the actual "job" is not finished. 