2/02/15
===
- Plotting in R studio 
- Undestand how the ratios are calculated


3/02/15
===

- Recalculating ratios. Instead of calculating the ratios per divisions, I am trying to calculate the ratios per each fragment. 
- Remove ratios that are lower than a given value (I am currently using 1). Then, I am trying to redefine the peak finding by only using these ratios (obviously, instead of using the whole genome, I have to define the search to the part of the genome where these ratios are calculated). 

4/02/15
===
Work on next week meeting


5/02/15
===
- Work on meeting
- When removing the contigs that are not involved in the causative mutation, the prediction is suddenly really bad because the SNP positions are calculated from the beginning, so the distribution is completely deviated. Therefore, I created a hash that relates the fragment and the SNP positions at each given fragment so I can assign the same positions to the fragments ordered with my SD method. However this doesn't work. 

6/02/15
===
Now, instead of using the whole genome length to predict the causal mutation I used the size of the genome where the hm SNP density is significant. I take out the fragments that have a hm to ht ratio lower than or equal to 1, so I only have the fragments were the ratio is higher, so the fragments were our causal mutation is located. 

```
contig_size = (genome_length/perm_hm.length).to_f

center = contig_size*(short_sh.length)

```
short_sh is the array of the fragments with the ratio is higher than 1. 
perm_hm is the initial array of fragments ordered with my method. 

Even though the SNPs densities are fine (see below), the location of the causal mutation cannot be determined with my current script. I am trying to work on this. 
####Correctly ordered genome
![Image](https://github.com/pilarcormo/small_genomes_SNPs/blob/master/Rplot.short_ok.png?raw=true)
####SDM ordered genome
![Image](https://github.com/pilarcormo/small_genomes_SNPs/blob/master/Rplot.short_sh.png?raw=true)

Fixed! So now, I am using this: 

```
mutation = Mutation.define(hm_list, ht_list, positions_hm, het_snps, center, ratios, expected_ratios) 
```
hm and ht list are the SNP positions for the correctly ordered genome. With them, we can calculate the ratios. Then, het_snps is the array with all the ht SNP positions in the SDM ordered genome while positions_hm is the array with the SNP positions in the group of fragments that have a high ratio. With them, we calculate the expected_ratios


####Next

- Work on SDM distribution methods, beucause even though the SNP identification is almost perfect, the hyp plot doesn't look very good. 

- Write text units for new approach of only considering the middle of the SNP distribution

- Use benchmark with all the steps, not only SDM