21/04/15
===

- Running alignments with mob mutants (bwa mem, TAIR10 as reference genome). Used reads from clean_data, skip trimming step. 
	- mob2-5: 109 
	- mob22-4: 62 
	
- Still running fph1 sampe. 
- Use mur vcf output (done with bwa mem even though the reads were short) to see the SNP distribution. There are not enough hm SNPs probably due to the reads that were not considered in the analysis.
- qqplot for OCF2 chromosome 2 after parental filtering. Take only the hm SNPs after the centromere (8 MB on). The correlation is very good. Write results (qqplots)